File: Gestion_Spring_Back_end/src/main/java/com/odji/spring_back_end/dto/DemandeDto.java
package com.odji.spring_back_end.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class DemandeDto {

    private Integer id;
    private String motif;

    // Relations sortantes exposées (sans suffixe "Dto")
    private LigneDemandeDto ligneDemande;
    private BureauDto bureau;
}

File: gestion-ressources-ng22/src/app/core/models/demande-dto.ts
import { PersonelDto } from '..\../core/models/personel-dto';
import { BureauDto } from '..\../core/models/bureau-dto';

export class DemandeDto {
  id: number= 0
  motif: string= "";

bureauDto: BureauDto= new BureauDto();
personelDto: PersonelDto= new PersonelDto();
}
